contracts/cw-core/src/contract.rs

```diff
@@ -182,37 +190,76 @@ pub fn execute_proposal_hook(
         .add_messages(msgs))
 }
 
-pub fn execute_update_admin(
+pub fn execute_nominate_admin(
     deps: DepsMut,
     sender: Addr,
-    admin: Option<Addr>,
+    nomination: Option<String>,
 ) -> Result<Response, ContractError> {
+    let nomination = nomination.map(|h| deps.api.addr_validate(&h)).transpose()?;
+
     let current_admin = ADMIN.load(deps.storage)?;
+    if current_admin.as_ref() != Some(&sender) {
+        return Err(ContractError::Unauthorized {});
+    }
 
-    match current_admin {
-        Some(current_admin) => {
-            // Check sender is the DAO Admin
-            if sender != current_admin {
-                return Err(ContractError::Unauthorized {});
-            }
+    let current_nomination = NOMINATED_ADMIN.may_load(deps.storage)?;
+    if current_nomination.is_some() {
+        return Err(ContractError::PendingNomination {});
+    }
 
-            // Save the new DAO Admin (which may be set to None)
-            ADMIN.save(deps.storage, &admin)?;
+    match &nomination {
+        Some(nomination) => NOMINATED_ADMIN.save(deps.storage, nomination)?,
+        None => ADMIN.save(deps.storage, &None)?,
```

Comment:
When thinking about the non-dao-dao ui usecase, Im a little worried that people could accidentally forget to fill in the `nomination` string with the address and then inadvertently just remove the admin from the dao with no recourse.

I think being explicit about removing the admin with a bool flag or something would be safer.

Reply:
agreed, we should make it difficult to footgun by removing the admin